bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/handler/MiIoLumiHandler.java

```diff
@@ -0,0 +1,157 @@
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.miio.internal.MiIoBindingConfiguration;
+import org.openhab.binding.miio.internal.basic.BasicChannelTypeProvider;
+import org.openhab.binding.miio.internal.basic.MiIoBasicDevice;
+import org.openhab.binding.miio.internal.basic.MiIoDatabaseWatchService;
+import org.openhab.binding.miio.internal.cloud.CloudConnector;
+import org.openhab.core.i18n.LocaleProvider;
+import org.openhab.core.i18n.TranslationProvider;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.type.ChannelTypeRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MiIoLumiHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Marcel Verpaalen - Initial contribution
+ */
+@NonNullByDefault
+public class MiIoLumiHandler extends MiIoBasicHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(MiIoLumiHandler.class);
+    private @Nullable MiIoGatewayHandler bridgeHandler;
+
+    public MiIoLumiHandler(Thing thing, MiIoDatabaseWatchService miIoDatabaseWatchService,
+            CloudConnector cloudConnector, ChannelTypeRegistry channelTypeRegistry,
+            BasicChannelTypeProvider basicChannelTypeProvider, TranslationProvider i18nProvider,
+            LocaleProvider localeProvider) {
+        super(thing, miIoDatabaseWatchService, cloudConnector, channelTypeRegistry, basicChannelTypeProvider,
+                i18nProvider, localeProvider);
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+        isIdentified = false;
+
+        final MiIoBindingConfiguration config = this.configuration;
+        if (config != null && config.deviceId.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Missing required deviceId");
+            logger.info("Missing required deviceId for {} {}", getThing().getUID(), getThing().getLabel());
+            return;
+        }
+        Bridge bridge = getBridge();
+        if (bridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "No device bridge has been configured");
+            logger.info("Missing Bridge for {} {}", getThing().getUID(), getThing().getLabel());
+            return;
+        } else {
+            logger.info("Bridge for {} {} = {} {} ({})", getThing().getUID(), getThing().getLabel(),
+                    bridge.getBridgeUID(), bridge.getLabel(), bridge.getHandler());
+        }
+
+        bridgeHandler = null;
+        if (ThingStatus.ONLINE != bridge.getStatus()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            return;
+        }
```

Comment:
Actually, this is done by the framework. To make this work, the Thing status must not be INITIALIZING. You could set it to UNKNOWN at the top of `initialize()`.

Reply:
Yes, seems indeed to be working fine as well...